src/common/snippets/src/lowered/pass/insert_tail_loop.cpp

```diff
@@ -173,37 +336,24 @@ bool InsertTailLoop::run(LinearIR& linear_ir) {
         if (!loop_end)
             continue;
 
+        const auto loop_info = loop_manager->get_loop_info(loop_end->get_id());
+        if (loop_info->fst_iter_handler) {
+            modified |= loop_info->fst_iter_handler(linear_ir, expr_it);
```

Comment:
Thanks for the finding. first iter handler for K block inserted the loop copy **after** `expr_it`. Because of that I always used `continue` right after first iter handler. However, I don't think that this was a right approach: in case of specific first iteration we need to insert it before the main body. I changed the handler in brgemm blocking pass and removed `continue;` from here,

Regarding handling return value: you are right, the return value indicates whether the LIR was changed by the handler or not